crates/orderbook/src/database/fee_policies.rs

```diff
@@ -51,16 +59,24 @@ impl super::Postgres {
             .collect::<HashMap<_, _>>();
         timer.stop_and_record();
 
-        fee_policies
-            .into_iter()
-            .map(|((auction_id, order_uid), policies)| {
-                policies
-                    .into_iter()
-                    .map(|policy| fee_policy_from(policy, quotes.get(&order_uid), order_uid))
-                    .collect::<anyhow::Result<Vec<_>>>()
-                    .map(|policies| ((auction_id, order_uid), policies))
-            })
-            .collect::<anyhow::Result<HashMap<_, _>>>()
+        let mut result = HashMap::new();
+        for (key, executed_fees) in executed_protocol_fees {
+            if let Some(policies) = fee_policies.get(&key) {
+                for (policy, executed_fee) in policies.iter().zip(executed_fees) {
+                    let executed_protocol_fee = ExecutedProtocolFee {
+                        amount: big_decimal_to_u256(&executed_fee.amount)
+                            .context("executed fee amount")?,
+                        token: primitive_types::H160(executed_fee.token.0),
+                        policy: fee_policy_from(policy.clone(), quotes.get(&key.1), key.1)?,
+                    };
+                    result
+                        .entry(key)
+                        .or_insert_with(Vec::new)
+                        .push(executed_protocol_fee);
+                }
+            }
+        }
+        Ok(result)
```

Comment:
Not related to this PR, just an observation that we have some inconsistency in the code style across the project where @m-lord-renkse tends to convert all the for loops into iterator transformation chains, and here we see the opposite.